### ISSUE TITLE
electron: 'open trace' command does not work properly

### ISSUE THREAD

**Issue by vince-fugnitto** [theia-ide/theia-trace-extension#227]
**Description**

The `open trace` command does not work successfully when starting the desktop application (`electron-app`).
The following error is thrown and can be viewed from the dev-tools console:

```
Uncaught (in promise) Error: Illegal props. Cannot have 'canSelectFiles' and 'canSelectFolders' at the same times. Props was: {"title":"Open Trace","canSelectFolders":true,"canSelectFiles":true}.
    at Object.toDialogProperties (/home/evinfug/worksp…ib/67.bundle.js:500)
    at ElectronFileDialogService.push.../node_modules/@theia/filesystem/lib/electron-browser/file-dialog/electron-file-dialog-service.js.ElectronFileDialogService.toOpenDialogOptions (/home/evinfug/worksp…ib/67.bundle.js:467)
    at ElectronFileDialogService.<anonymous> (/home/evinfug/worksp…ib/67.bundle.js:370)
    at step (/home/evinfug/worksp…ib/67.bundle.js:305)
    at Object.next (/home/evinfug/worksp…ib/67.bundle.js:286)
    at fulfilled (/home/evinfug/worksp…ib/67.bundle.js:277)
```

The error is due to the open props by setting both `canSelectFolders` and `canSelectFiles` as options:

https://github.com/theia-ide/theia-trace-extension/blob/3908ad2aef7a86ca20a0348a41045ed5801bfbf7/viewer-prototype/src/browser/trace-viewer/trace-viewer-contribution.ts#L38-L43

**Steps to reproduce**

1. start the electron example application: `yarn rebuild:electron && (cd electron-app && yarn start)`
2. open the quick-command palette: <kbd>ctrl</kbd>+<kbd>shift</kbd>+<kbd>p</kbd>
3. attempt to execute `open trace` - nothing happens and the error is thrown
4. alternatively you can open the `trace explorer` and click the `open trace` button

Reference: https://github.com/eclipse-theia/theia/blob/b6add4a6e6e97cbdff73279fab0a0b116576ccb5/packages/filesystem/src/electron-browser/file-dialog/electron-file-dialog-service.ts#L143-L154

**Comment by bhufmann** [comment]
@vince-fugnitto that's a problem for us. A trace can be a single for or a directory with with multiple files (CTF traces). It's a important to be able to select either a file or directory. Moreover, when user opens a directory, then the code is checking if there one or more CTF traces in it. 

So, the trace extension, it's an important feature. We had issues in Eclipse with having either a file or directory dialog only, but not both. I thought we would have that issues in the new platform.

Anyways, any ideas on how to fix that?

**Comment by vince-fugnitto** [comment]
> Anyways, any ideas on how to fix that?

@bhufmann it’s a limitation from Electron itself that on Linux and Windows the dialog cannot be for both files and folders simultaneously: 

> Note: On Windows and Linux an open dialog can not be both a file selector and a directory selector, so if you set properties to ['openFile', 'openDirectory'] on these platforms, a directory selector will be shown.

Perhaps for the browser you can have the dialog accept both, and for electron you fallback to two different dialogs (one file and one folder) like you had previously? Else, I was thinking that we somehow customize the behaviour to use the browser type dialog even in the electron (desktop) application.

**Comment by vince-fugnitto** [comment]
I haven’t verified the behaviour yet, but I wonder if the browser version supports opening both files and folders simultaneously.

**Comment by bhufmann** [comment]
> I haven’t verified the behaviour yet, but I wonder if the browser version supports opening both files and folders simultaneously.

In the browser version you can select either a file or directory from the same dialog. We don't use multi-selection at the moment.

**Comment by bhufmann** [comment]
I think, for now we should only open the directory chooser, because the current main use case are Common Trace Format (CTF) traces analyzed by a Trace Compass server. CTF traces are directories (metadata file and binaries). Using the directory chooser we will be able to open a single CTF trace as well as a set of CTF traces by selecting a the parent directory with multiple CTF traces. This is for the MVP. Single file traces can still be opened from the file explorer using "Open With -> Trace Viewer"

In the future, we'll need to consider opening single file traces together with CTF traces and this will be handled in a different issue tracker.

**Comment by vince-fugnitto** [comment]
> I think, for now we should only open the directory chooser...

@bhufmann sounds good, I was also going to suggest as a future usability improvement to allow traces to be drag-and-dropped into the explorer as well (similarly to how the `explorer/navigator` can accept files/foders to be drag-and-dropped).

**Comment by bhufmann** [comment]
Please note that in the future there needs to be a way to detect traces (files) and add them to an experiment (set of traces analyzed) together. That was not the scope of fix #232.
